Dockerfile

```diff
@@ -47,10 +51,26 @@ RUN mvn clean install -T1C -U -f /src/dhis-2/dhis-web/pom.xml -DskipTests
 ##########
 FROM tomcat:8.5.34-jre8-alpine as serve
 
-RUN rm -rf /usr/local/tomcat/webapps/*
+COPY docker-entrypoint.sh /usr/local/bin/
+
+RUN rm -rf /usr/local/tomcat/webapps/* && \
+    mkdir /usr/local/tomcat/webapps/ROOT && \
+    chmod +rx /usr/local/bin/docker-entrypoint.sh && \
+    mkdir /DHIS2_home && \
+    addgroup -S tomcat && \
+    addgroup root tomcat && \
+    adduser -S -D -G tomcat tomcat
+
+RUN apk add --update --no-cache \
+        bash  \
+        su-exec
 
 COPY server.xml /usr/local/tomcat/conf
 COPY --from=build /src/dhis-2/dhis-web/dhis-web-portal/target/dhis.war /usr/local/tomcat/webapps/ROOT.war
 
-# Expose the easy-to-remember directory /DHIS2_home for Docker volume mounting to configure the CORE instance
 ENV DHIS2_HOME=/DHIS2_home
```

Comment:
One tiny comment, would it be possible to extract this location (`/DHIS2_home`) out into a buildvar?  Otherwise (and perhaps simpler) if we move `ENV DHIS2_HOME=/DHIS2_home` then we can reference that variable when making the directory (adding -p = `mkdir -p $DHIS2_HOME`)

Just a thought, not crucial.  The buildvar might make things a touch more extensible, though our official images will still use the straightforward `/DHIS2_home` directory.

Reply:
very good point, done!